### PR TITLE
doc: add examples to xline client

### PR DESCRIPTION
**Pull request by bsbds**
This PR is based on #373 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    descibed in `Add examples to the crate` section in #369 

* what changes does this pull request make?

   Add use examples to xline client

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: xline-client/examples/auth.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    let client = Client::connect(curp_members, ClientOptions::default())
+        .await?
+        .auth_client();
+
+    // enable auth
+    let _resp = client.auth_enable().await?;
+
+    // connect using the root user
+    let options = ClientOptions::default().with_user("root", "rootpwd");
+    let client = Client::connect(curp_members, options).await?.auth_client();
```
Comment (Phoenix500526) [inline xline-client/examples/auth.rs:21]: Do we have to reconnect the cluster once we enable auth?

**Review by Phoenix500526 — COMMENTED** [review]
Please use `debug!` or `info!` to print logs. And we should add some error handling examples in this pr.

File: xline-client/src/clients/auth.rs [hunk 2]
```diff
@@ -66,6 +66,31 @@ impl AuthClient {
...
+    ///     let client = Client::connect(curp_members, ClientOptions::default())
+    ///         .await?
+    ///         .auth_client();
+    ///
+    ///     let _resp = client.auth_enable().await?;
+    ///
+    ///     // auth with some user
+    ///
+    ///     Ok(())
+    /// }
```
Comment (themanforfree) [inline xline-client/src/clients/auth.rs:92]: Add main function to comment example is redundant, FYI: [Rust Std](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by_key).
The other comments too.

File: xline-client/examples/auth.rs (hunk 1 above)
Comment (bsbds) [reply xline-client/examples/auth.rs:21]: yes, currently the auth token is acquired during the connect stage

File: xline-client/examples/auth.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+use xline_client::{error::Result, Client, ClientOptions};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // the name and address of all curp members
+    let curp_members = [
```
[the remark this answers is not in the document]
Comment (bsbds) [reply xline-client/examples/auth.rs:6]: in the examples folder it doesn't seems possible to create a `common` module

File: xline-client/src/clients/auth.rs (hunk 2 above)
Comment (bsbds) [reply xline-client/src/clients/auth.rs:92]: The main function is required here as the code need a runtime to compile and also need to return error.

**Review by iGxnon — DISMISSED** [review]

**Review by themanforfree — DISMISSED** [review]

File: xline-client/examples/error_handling.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    // We try to update the key using its previous value.
+    // It should return an error and it should be `key not found`
+    // as we did not add it before.
+    let resp = client
+        .put(PutRequest::new("key", "").with_ignore_value(true))
+        .await;
+    let err = resp.unwrap_err();
+
+    // We first match the client error
+    let ClientError::ProposeError(pe) = err else { unreachable!("should be a propose error")};
```
Comment (Phoenix500526) [inline xline-client/examples/error_handling.rs:32]: Error message should be "XXXX should be err after compaction". XXXX stands for some actions.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by themanforfree — APPROVED** [review]

**Pull request merged**